SPEC.md

```diff
@@ -215,11 +249,26 @@ An example trace:
 
 ```
 
-## Versioning
+### Resource restrictions
+
+In order to ensure analysis runs reliably across a variety of systems, Engines
+must conform to some basic resource restrictions:
+
+* The Docker image for an Engine must not exceed 512MB, including all layers.
+* The combined total RSS memory usage by all processes within the Docker container must not exceed 1GB at any time.
+* All Engines must complete and exit within 10 minutes.
 
-This specification is versioned. The current version is [in the repository](https://github.com/codeclimate/spec/blob/master/VERSION). Engines declare the version of the specification they are compatible with in the manifest file, described below. This project follows [Semantic Versioning](http://semver.org/).
+### Security restrictions
 
-## Engine Specification File
+Engines run in a secured runtime environment, within container-based virtualization
+provided by Docker.
+
+* The root filesystem (`/`) is mounted read-only. A `/tmp` volume is mounted read-write for temporary file storage during the engine run.
```

Comment:
I didn't actually change any of the text here, just placed in a different section to make it clearer that these details are mostly relevant to execution of the main engine `CMD`. You can see in this in master here: https://github.com/codeclimate/spec/blob/master/SPEC.md#security-restrictions

I'm not familiar enough with these details to offer a good correction, but if you've got some text I could drop I'd be happy to include it in this PR>